File: scala-example/src/main/scala/thirtytwo/degrees/halfpipe/scalaexample/Context.scala
package thirtytwo.degrees.halfpipe.scalaexample

import org.springframework.context.annotation._
import com.netflix.config.scala.DynamicProperties
import javax.inject.Named
import org.springframework.context.annotation.ComponentScan.Filter
import org.springframework.stereotype.Controller
import thirtytwo.degrees.halfpipe.DefaultScalaContext
import thirtytwo.degrees.halfpipe.context.MetricsContext
import thirtytwo.degrees.halfpipe.mgmt.resources.GCResource

@Configuration
@ComponentScan (basePackageClasses = Array (classOf[Context], classOf[GCResource]),
  excludeFilters = Array(new Filter (Array (classOf[Controller], classOf[Configuration]))))
@Import(Array(classOf[DefaultScalaContext], classOf[SecurityContext], classOf[MetricsContext]))
class Context extends DynamicProperties {

  @Bean @Named("helloText")
  def helloText() = dynamicStringProperty("hello.text", "Hello default")

  @Bean
  def garbageCollectionTask: GCResource = new GCResource
}

File: halfpipe-example/src/main/java/halfpipe/example/Context.java
package halfpipe.example;

import org.springframework.context.annotation.*;
import org.springframework.context.annotation.ComponentScan.Filter;
import org.springframework.stereotype.Controller;
import halfpipe.context.DefaultContext;
import halfpipe.context.MetricsContext;
import halfpipe.mgmt.resources.GCResource;

/**
 * User: spencergibb
 * Date: 9/21/12
 * Time: 4:18 PM
 * <p/>
 * http://static.springsource.org/spring/docs/3.1.x/spring-framework-reference/html/mvc.html#mvc-config-enable
 */
@Configuration
@ComponentScan(basePackageClasses = {Context.class, GCResource.class}, excludeFilters = {
    @Filter(Controller.class),
    @Filter(Configuration.class)
})
//@ImportResource("classpath:META-INF/spring/applicationContext-security.xml")
@Import({DefaultContext.class, MetricsContext.class, SecurityContext.class})
public class Context {

    @Bean
    public GCResource garbageCollectionTask() {
        return new GCResource();
    }

}
